test/e2e/e2e_test.go

```diff
@@ -43,34 +43,64 @@ import (
 // in state deleting
 
 var _ = Describe("controller", Ordered, func() {
-	var namespace string = "instaslice-operator-system"
-	var defaultNamespace string = "default"
+	var (
+		namespace         string = "instaslice-operator-system"
+		defaultNamespace  string = "default"
+		isRunningOnOCP    bool
+		kubeCli           = "kubectl"
+		ocpCli            = "oc"
+		dockerBin         = "docker"
+		podmanBin         = "podman"
+		deployEmulated    = "deploy-emulated"
+		deployOCPEmulated = "ocp-deploy-emulated"
+		clientBin         = kubeCli
+		containerTool     = dockerBin
+		deployEmulatedArg = deployEmulated
+	)
+	ocpMode := os.Getenv("OCP_MODE")
+	if ocpMode != "" {
+		isRunningOnOCP = true
+		clientBin = ocpCli
+		containerTool = podmanBin
+		deployEmulatedArg = deployOCPEmulated
+	}
 
 	BeforeAll(func() {
-		fmt.Println("Setting up Kind cluster")
-		cmd := exec.Command("kind", "create", "cluster", "--image", "kindest/node:v1.30.0@sha256:047357ac0cfea04663786a612ba1eaba9702bef25227a794b52890dd8bcd692e")
-		output, err := cmd.CombinedOutput()
-		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to create Kind cluster: %s", output))
+		if !isRunningOnOCP {
```

Comment:
@rphillips and I discussed this before.

This can be a followup but we should really aim to separate creation of clusters from the e2e code.

I think the e2e test should assume their is a k8s cluster with this installed. And we can change the deploy code via scripts.

I could see an e2e test where we test against Kind versus OCP and it would be great if we don't need to change the e2e test.

Reply:
> This can be a followup but we should really aim to separate creation of clusters from the e2e code.

+1 
I've raised this PR keeping in mind that we don't have a job that runs even the emulator mode today to test the code changes.
Ultimately, we should aim for the seperate creation of clusters, load images built from the fresh code etc.

Reply:
I think we could refactor this PR to create the kind cluster in the Makefile? Correct?

Reply:
Created an issue for now to track [here](https://github.com/openshift/instaslice-operator/issues/192)